src/features/counterfactual/ActivateAccountButton.tsx

```diff
@@ -25,7 +25,7 @@ const ActivateAccountButton = () => {
   return (
     <Tooltip title={isProcessing ? 'The safe activation is already in process' : undefined}>
       <span>
-        <CheckWallet allowNonOwner>
+        <CheckWallet allowNonOwner allowUndeployedSafe>
```

Comment:
Do we need CheckWallet here at all? Afaiu the only thing we need to make sure is that the user has any wallet connected. If we could ensure this without CheckWallet we wouldn't need another flag.

Reply:
But isn't that exactly what CheckWallet is for? So I don't need to use the `useWallet` hook here and worry about rendering a tooltip around the button etc?

Reply:
Yes, makes sense. I created a ticket #4210 to refactor this component.